### PR TITLE
Api updates

### PR DESCRIPTION
**Pull request by jsnoble**

### CODE REVIEW

File: lib/search.js [hunk 1]
```diff
@@ -5,38 +5,58 @@ var _ = require('lodash');
...
+            errResponse.criteria = query
+        }
+
+        res.status(statusCode).json(errResponse);
+    }
+
     function validate(res, data, type, errField) {
         if (typeof data !== type) {
-            res.status(500).json({error: "the " + errField + " parameter must be a " + type + ", was given: " + data});
+            res.status(400).json({error: "the " + errField + " parameter must be a " + type + ", was given: " + data});
```
Comment (kstaken) [inline lib/search.js]: Changing these response codes is a breaking change which can't be accepted.

**Comment by jsnoble** [comment]
made the changes

File: lib/search.js [hunk 2]
```diff
@@ -5,14 +5,32 @@ var _ = require('lodash');
...
+        if (message) {
+            errResponse.message = message
+        }
+
+        if (error) {
+            errResponse.error = error
+        }
+
+        if (debug) {
+            errResponse.criteria = query
```
Comment (kstaken) [inline lib/search.js]: This is just parroting back what the user submitted which isn't what the ticket specified. This was only concerned with ES search errors NOT all errors. The goal here is to show the query as it's being sent to ES and the ES error response.

**Comment by jsnoble** [comment]
closed due to security concerns

**Pull request closed without merging**